python/ray/util/sgd/v2/backends/torch.py

```diff
@@ -137,6 +139,13 @@ def set_env_vars(addr, port):
 
     def on_shutdown(self, worker_group: WorkerGroup,
                     backend_config: TorchConfig):
-        if len(worker_group) > 1:
-            worker_group.execute(dist.destroy_process_group)
-        worker_group.execute(shutdown_torch)
+
+        worker_group.execute(
+            shutdown_torch, destroy_process_group=len(worker_group) > 1)
+
+    def handle_failure(self, worker_group: WorkerGroup,
+                       failed_worker_indexes: List[int],
+                       backend_config: BackendConfig):
+        worker_group.shutdown()
+        worker_group.start()
+        self.on_start(worker_group, backend_config)
```

Comment:
Can you document the torch-specific reason for this logic? What about torch requires us to shutdown the workers entirely?

Reply:
I added a comment for this. 

Basically, `torch` does not allow a process group to be created multiple times on the same process. So we have to kill all the workers and restart them again. This is exactly what Torch Elastic does as well.